### ISSUE TITLE
Implement Yolo based on PANet ~+4 AP@[.5, .95]

### ISSUE THREAD

**Issue by AlexeyAB** [AlexeyAB/darknet#3175]
Implement Yolo based on PANet: https://arxiv.org/abs/1803.01534v4

1. There is added very short **Bottom-up Path Augmentation** - so there are added shortcuts (or conctats) to FPN (Feature Pyramid Networks for Object Detection that is used for Yolo v3), so each of 4 final layers has no more than 10 intermediate layers from any Level.

2. Also there is added **Adaptive Feature Pooling Structure**
    * either just 3 [shortcut] layers, for element-wise adding of 4 final layers
    * or we can implement pooling layer that uses max operation across channels of two input layers

3. May be use both Deconvolutional layer and Fully-connected layer as in PANet, or just Deconvolutional layer and YOLACT (You Only Look At CoefficienTs) https://github.com/AlexeyAB/darknet/issues/3048

![image](https://user-images.githubusercontent.com/4096485/57894068-15cc3f00-784e-11e9-98ab-bf85781dab8d.png)

![image](https://user-images.githubusercontent.com/4096485/57894054-077e2300-784e-11e9-9eea-21fe9b7b2afc.png)

![image](https://user-images.githubusercontent.com/4096485/57939086-2faf6580-78d2-11e9-9107-936cbc4783b8.png)

----

* PANet (Path Aggregation Network for Instance Segmentation): https://arxiv.org/abs/1803.01534v4
* FPN (Feature Pyramid Networks for Object Detection): https://arxiv.org/abs/1612.03144v2

**Comment by AlexeyAB** [comment]
Some form of PANet implementation: https://github.com/AlexeyAB/darknet/issues/3114#issuecomment-494148968

**Comment by AlexeyAB** [comment]
Example:

cfg-file: https://github.com/AlexeyAB/darknet/files/3253820/yolo_v3_spp_pan_scale.cfg.txt

pre-trained weights-file for training: http://pjreddie.com/media/files/darknet53.conv.74

training command: `./darknet detector train data/self_driving.data yolo_v3_spp_pan_scale.cfg.txt darknet53.conv.74 -map`

detection command: `./darknet detector demo data/self_driving.data yolo_v3_spp_pan_scale.cfg.txt backup/yolov3-tiny_occlusion_track_last.weights self_driving_valid.avi -out_filename out_self_driving_valid.avi`

result training weights-file: https://drive.google.com/file/d/14557YbnxPepJIUlrTdaphhNmMAVLsOqq/view this dataset: https://drive.google.com/open?id=1QzXSCkl9wqr73GHFLIdJ2IIRMgP1OnXG

----

example of detection: https://drive.google.com/open?id=1EcHfzgypfk57pQlLl379c-CKvSy5l3zg

example of training Loss & mAP chart:

![chart](https://user-images.githubusercontent.com/4096485/59544198-b3339500-8f18-11e9-8ceb-f6e29f4ca5fb.png)

**Comment by Shengxiaolin** [comment]
hi @AlexeyAB i have 2 questions about PANet model:
(1)do i need change anchors and masks parameters to fit my own dataset or use the default anchors is enough(i use it to detect helmet) .and the modify rules is same with other cfg file?
(2)max_batches parameter also equals 2000*classes?
looking forward to your reply :)
